drivers/usbc/tcpc/ucpd_stm32.c

```diff
@@ -1009,46 +1009,58 @@ static void ucpd_isr(const struct device *dev_inst[])
 	const struct tcpc_config *config;
 	struct tcpc_data *data;
 	uint32_t sr;
-	uint32_t ucpd_base;
 	struct alert_info *info;
 	uint32_t tx_done_mask = UCPD_SR_TXMSGSENT |
 				UCPD_SR_TXMSGABT |
 				UCPD_SR_TXMSGDISC |
 				UCPD_SR_HRSTSENT |
 				UCPD_SR_HRSTDISC;
 
-	if (IS_ENABLED(CONFIG_SOC_SERIES_STM32G0X)) {
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) > 1
+#ifdef CONFIG_SOC_SERIES_STM32G0X
+	/*
+	 * Multiple UCPD ports are available
+	 */
+
+	uint32_t ucpd_base;
+
+	/*
+	 * Since the UCPD peripherals share the same interrupt line, determine
+	 * which one generated the interrupt.
+	 */
+	if (LL_SYSCFG_IsActiveFlag_UCPD1()) {
+		/* UCPD1 interrupt is pending */
+		ucpd_base = UCPD1_BASE;
+	} else if (LL_SYSCFG_IsActiveFlag_UCPD2()) {
+		/* UCPD2 interrupt is pending */
+		ucpd_base = UCPD2_BASE;
+	} else {
 		/*
-		 * Since the UCPD peripherals share the same interrupt line, determine
-		 * which one generated the interrupt.
+		 * The interrupt was triggered by some other device sharing this
+		 * interrupt line.
 		 */
-		if (LL_SYSCFG_IsActiveFlag_UCPD1()) {
-			/* UCPD1 interrupt is pending */
-			ucpd_base = UCPD1_BASE;
-		} else if (LL_SYSCFG_IsActiveFlag_UCPD2()) {
-			/* UCPD2 interrupt is pending */
-			ucpd_base = UCPD2_BASE;
-		} else {
-			/*
-			 * The interrupt was triggered by some other device sharing this
-			 * interrupt line.
-			 */
-			return;
-		}
+		return;
+	}
 
-		/* Find correct device instance for this port */
-		for (int i = 0; i < DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT); i++) {
-			dev = dev_inst[i];
-			config = dev->config;
-			if ((uint32_t)(config->ucpd_port) == ucpd_base) {
-				break;
-			}
-		}
-	} else {
-		/* Only one port available */
-		dev = dev_inst[0];
+	/* Find correct device instance for this port */
+	for (int i = 0; i < DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT); i++) {
+		dev = dev_inst[i];
 		config = dev->config;
+		if ((uint32_t)(config->ucpd_port) == ucpd_base) {
+			break;
+		}
 	}
+#else
+#error "Multiple UCPD ports defined for device with only one port."
+#endif /* CONFIG_SOC_SERIES_STM32G0X */
```

Comment:
This is not required. Device tree parsing will prevent this to happen.
The `#ifdef CONFIG_SOC_SERIES_STM32G0X` could be safely removed.

Reply:
Done